### ISSUE TITLE
File names dont match with imports in JS files when building --prod

### ISSUE THREAD

**Issue by obrunopolo** [ionic-team/stencil#2687]
**Stencil version:**

```
 @stencil/core@2.0.3
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

The app works well while using `stencil build --serve --dev --watch`, but once I build a production version, run a http server on the `www` folder and open it in the browser, it gets stuck on white screen on 50% of builds. On console I could see error 404, some resources were missing.

**Expected behavior:**

Production builds should work reliably.

**Steps to reproduce:**

Create a "Starter app" template and run `npm start` and `npm run build` a few times. After each production build, try to open on browser using some http server.

In my case I could see that index.html loads the first hashed js file successfully, but then it try to import 2 non-existing files.

Please note that this may not be related to the hash of file names, because I could reproduce this behavior with `hashFileNames: false` . In this case, it was importing a file with different content hash.

**Comment by MarkChrisLevy** [comment]
I can confirm that issue. It is also not related to caching (both in browser and cli "--no-cache" setting).

**Comment by obrunopolo** [comment]
@MarkChrisLevy that's correct. I also tryied to use `enableCache: false`  and still get the same error.

Just got this again today:

![image](https://user-images.githubusercontent.com/47830258/95240838-58940b80-07e3-11eb-9949-72e69c8c0407.png)

**Comment by r3dDoX** [comment]
I'm having the same Problem since v2. Opened another issue couple weeks ago: #2671

**Comment by Scan0815** [comment]
i can also reproduce this on 2.2.0 stenciljs

**Comment by r3dDoX** [comment]
I am still able to reproduce this with 2.3.0

**Comment by Scan0815** [comment]
@adamdbradley , 
I'm sorry to speak to you directly, but the error is really annoying, having to constantly delete the build folder in a running project. in version 1 it worked great.

**Comment by MrAntix** [comment]
you can reproduce this with the starter app

``` 
npm init stencil 
npm i

npm run build => built index.html still has the app.css and app.js files in the head
npm run build => built index.html has compiled references except the nomodule one
```

**Comment by kevinlandsberg** [comment]
Can reproduce the error in stencil 2.3.0
Always getting 404's for bundle file inside www

**Comment by rohanrp** [comment]
A temp fix is to delete the dist and www folders before running the build command. e.g

npm install --save-dev del-cli

package.json:
 "build": "**del-cli --force ./www && del-cli --force ./dist &&** stencil build --prod",

**Comment by theo-staizen** [comment]
> A temp fix is to delete the dist and www folders before running the build command. e.g
> 
> npm install --save-dev del-cli
> 
> package.json:
> "build": "**del-cli --force ./www && del-cli --force ./dist &&** stencil build --prod",

Our npm script was already doing that (using rimraf, but does same thing) and it doesn't help

**Comment by MrAntix** [comment]
you can also use the empty flag in config

```  outputTargets: [
    {
      type: 'www',
      dir: 'wwwroot',
      empty: true
    },
```

**Comment by r3dDoX** [comment]
Just tried today with 2.4.0 and I can still reproduce this bug.

**Comment by alicewriteswrongs** [comment]
Hey all! I just merged a change (#4992) which I believe will fix this issue. That fix will be included in our next Stencil release, so stay tuned for that! I (or someone else on the team) will ping here when that's ready, so stay tuned 🙂 I'm going to close for now
